### DISCUSSION TITLE
Cron job runs 6 times

### DISCUSSION THREAD

**Discussion by FuzzyMistborn** [woodpecker-ci/woodpecker#3777] [Ask a Question] [answered]
Figure this isn't a bug with Woodpecker but rather a me problem (switching from Drone to Woodpecker).  I have a pipeline running on cron, and it ran last night but seems to have been triggered 6 times.  Should have only been triggered once.

![image](https://github.com/woodpecker-ci/woodpecker/assets/16689090/581f5d16-1f1f-4c9d-8913-a32a1456900e)

Here are the steps (it's kludgey but it works):

```
steps:
  - name: Check Dockerhub Nextcloud Version
    image: fuzzymistborn/docker-alpine-skopeo:latest
    commands:
      - /bin/sh ./update.sh

  - name: push
    image: drone/git
    secrets:
      - gh_api_key
      - email
      - tg_id
      - tg_token
    commands:
      - |+
        if [ $(git diff HEAD | wc -l) -gt 1 ]
        then
        git config --global user.email "$email"
        git config --global user.name "FuzzyMistborn"
        git add .
        git remote add deploy https://fuzzymistborn:$gh_api_key@github.com/FuzzyMistborn/nextcloud-docker.git
        DOCKER_VER=$( cat docker_ver )
        git commit -m "[CI Skip] Update Docker image to $DOCKER_VER"
        git tag -a $DOCKER_VER -m "Nextcloud Release v. $DOCKER_VER"
        git push deploy main
        git push deploy main $DOCKER_VER
        curl -s -X POST https://api.telegram.org/bot$TG_TOKEN/sendMessage -d chat_id=$TG_ID -d parse_mode=markdown -d text="✅  New Nextcloud Docker Release! %0A🌐  [Output]({{build.link}})"
        git remote rm deploy
        fi

  - name: Send TG Notification
    image: appleboy/drone-telegram
    settings:
      token:
        from_secret: tg_token
      to:
        from_secret: tg_id
      format: markdown
      message: |-
        ❌  Nextcloud Docker Updater Task has *FAILED*!
        🌐  [Output]({{build.link}})
    when:
      event: {}
      status:
        - failure

when:
  event: cron
  cron: "update"
  ```

**Comment by qwerty287** [comment]
This looks like a bug to me, I'm not aware of how you could trigger a cron multiple times.
What's the configuration of your cron? And is there something interesting in the logs?

We had a similar issue with PRs but not sure if that's related.

**Reply by FuzzyMistborn** [reply]
Ok good to know I'm not crazy then hahaha.

Cron is  `* * */6 * * *` (running every 6 hours).  I didn't see anything crazy in the logs, but I did revert to a manual push.  I'll revert to cron and see if there's anything.  I don't recall seeing anything odd before though.

**Reply by zc-devs** [reply]
Reproduced in Woodpecker `2.5.0`, Kubernetes.

Pipeline:
```yaml
skip_clone: true
steps:
  server:
    image: alpine
    commands:
      - echo Hello
 ```

Cron:
```
* 1/5 * * * *
```

Repeated 7 times:
[Screenshot 1](https://github.com/woodpecker-ci/woodpecker/assets/127358482/299dcea9-891d-4bd3-9f8d-95f60ff2f488)
[Screenshot 2](https://github.com/woodpecker-ci/woodpecker/assets/127358482/b3f7959c-82df-4f8b-9923-c39e1d361569)

[woodpecker-server.log](https://github.com/user-attachments/files/15817468/woodpecker-server.log)
[woodpecker-agent.log](https://github.com/user-attachments/files/15817469/woodpecker-agent.log)

**Reply by qwerty287** [reply, marked as answer]
I'm not a cron expert, but we have seconds as the first char and doesn't `*` mean *on every second*?

Can you check what happens with `0 1/5 * * * *` or ` 0 * */6 * * *`?

**Reply by zc-devs** [reply]
`0 1/5 * * * *`

[Works as expected](https://github.com/woodpecker-ci/woodpecker/assets/127358482/3148cee4-18b6-41ee-9a7d-4ace729dbdf9)

> doesn't * mean on every second?

[You are right](https://github.com/woodpecker-ci/woodpecker/assets/127358482/e15e5bff-f6f7-4a18-9d94-199dba701290)


Something should definitely be improved in that space :)
#3781

**Reply by FuzzyMistborn** [reply]
Thank you all, I figured there was user error involved.  `0 0 */6 * * *` seems to be working.
